libsolidity/ast/Types.cpp

```diff
@@ -3139,13 +3149,15 @@ string FunctionType::externalSignature() const
 
 	// "inLibrary" is only relevant if this is not an event.
 	bool const inLibrary = kind() != Kind::Event && dynamic_cast<ContractDefinition const&>(*m_declaration->scope()).isLibrary();
-	FunctionTypePointer external = interfaceFunctionType();
```

Comment:
If `interfaceFunctionType` is not called anymore, we ignore the return parameters. Is that intended?

Reply:
Perhaps we should also change the implementation of  `interfaceFunctionType` to include the special case about events in the line above.

Reply:
Originally, the return value of `interfaceFunctionType()` was only used to get the `parameterTypes()`, so we basically did extra work for something we didn't even use eventually..

Regarding the special case: That was my original approach in this PR, @ekpyron suggested a different approach here: https://github.com/ethereum/solidity/pull/6473#issuecomment-480871065

Our fear was that it might have unforeseen effects on other functions using `interfaceFunctionType()`